custom_components/tahoma/tahoma_device.py

```diff
@@ -109,10 +113,26 @@ def device_info(self) -> Dict[str, Any]:
         manufacturer = self.select_state(CORE_MANUFACTURER_NAME_STATE) or "Somfy"
         model = self.select_state(CORE_MODEL_STATE) or self.device.widget
 
+        name = self.name
+        device_url = self.device_url
+        if "#" in self.device_url:
+            device_url = self.device.deviceurl.split("#", 1)[0]
+            entity_registry = self.hass.data["entity_registry"]
```

Comment:
A comment would be nice, because this part of code is not really readable IMO.

Reply:
done